memory/src/main/scala/filodb.memory/data/ChunkMap.scala

```diff
@@ -220,6 +220,19 @@ abstract class ChunkMap(var capacity: Int) {
     arrayGet(realIndex(first + size - 1))
   }
 
+
+  final def chunkmapDoGetLastIterator(): ElementIterator = {
```

Comment:
Can you add a unit test in `ChunkMapTest`  that exercises the code and validates locks ?

Reply:
Done, the code path is getting used in the new test and the fact that `chunkmapFree` at the end which acquires exclusive lock works validated that no locks are help.